src/main/java/be/syntra/devshop/DevshopBack/services/DataFillerServiceImpl.java

```diff
@@ -180,6 +185,35 @@ public void initialize() {
                             "admin@emaill.com")
             ));
         }
+
+        if (cartRepository.count() == 0) {
+            userRepository.findAll().forEach(this::setArchivedCartsAndSave);
+            userRepository.flush();
+        }
+    }
+
+    private void setArchivedCartsAndSave(User user) {
+        user.setArchivedShopOrders(List.of(createCart()));
+    }
+
+    private ShopOrder createCart() {
+        return shopOrderFactory.of(getListOfRandomProducts(), true, true, getCartContent());
+    }
+
+    private List<Product> getListOfRandomProducts() {
+        List<Product> products = new ArrayList<>();
+        for (int i = 0; i < 5; i++) {
+            products.add(productRepository.findAll().get(new Random().nextInt(productRepository.findAll().size())));
+        }
+        return products;
+    }
+
+    private List<OrderContent> getCartContent() {
```

Comment:
I understand the reluctancy to extract some of these methods to their factory classes. This facade encompasses a lot of the factories and repositories we need for populate the database. This way, we don't have to change the testClasses. However, something keeps bugging me, but after thinking it over, I also can't decide which to move #ohWell. 
I really love streams though, but that isn't a requisite. Just wanted to show another way of writing the loop. What DOES bother me a lot, is that you used two queries on one line to do the same thing. If you're not changing anything else, I would at least change to Random().nextInt(productRepository.count())

Reply:
damn, i think this is about the deleted method, so i don't know what its about now, BUT i do admit that i use streams waaaaay too little and need practice on streams altough i realy like them

Reply:
i did however renamed cart to shopOrder trough out this whole class now

Reply:
The cleanup was truly effective. Now it's easier to see the flow. Which means that it's now clear that the getShopOrderContent() method needs to move to the shopOrderFactory. Meaning that it can be removed as a shopOrderFactory.of() parameter.
When you did that, it's time to refactor the shopOrderFactory.of() even further. Your boolean parameters are hardcoded values, so the parameters can be removed, and the booleans directly set in the factorymethod itself.
That leaves you with all logic neatly encapsulated in the factory class and your shopOrderFactory.of() method will have NO parameters anymore and can be directly used in the setArchivedShopOrderAndSave(User user) method. That method's body now becomes:
user.setShopOrders(List.of(shopOrderFactory.of())

Reply:
nice! fixed

Reply:
had to rewrite test ShopOrderFactoryTest also